src/components/appointments/LocationAndTimeStep.js

```diff
@@ -24,15 +48,77 @@ const LocationAndTimeStep = () => {
         headingType="subtitle"
         modifiers={['padded']}
       />
+
+      <ProductSummary />
+
       <div>
         <LocationSelect />
         <DateSelect />
         <TimeSelect />
       </div>
+
+      <SubmitRow
+        canSubmit={isStepValid(values)}
+        nextText={intl.formatMessage({
+          description: 'Appointments location and time step: next step text',
+          defaultMessage: 'To contact details',
+        })}
+        previousText={intl.formatMessage({
+          description: 'Appointments location and time step: previous step text',
+          defaultMessage: 'Back to products',
+        })}
+        navigateBackTo="producten"
+      />
     </>
   );
 };
 
 LocationAndTimeStep.propTypes = {};
 
+const ProductSummary = () => {
+  const {baseUrl} = useContext(ConfigContext);
+  const {
+    loading,
+    value: allProducts,
+    error,
+  } = useAsync(async () => await getProducts(baseUrl), [baseUrl]);
+  const {
+    values: {products},
+  } = useFormikContext();
+
+  if (!products.length) return null;
+  if (error) throw error;
+  if (loading) {
+    return <Loader modifiers={['small']} />;
+  }
+
+  const productsById = Object.fromEntries(allProducts.map(p => [p.identifier, p.name]));
+  return (
+    <>
+      <Heading3 className="utrecht-heading-3--distanced">
+        <FormattedMessage
+          description="Product summary on appointments location and time step heading"
+          defaultMessage="Your products"
+        />
+      </Heading3>
+      <UnorderedList className="utrecht-unordered-list--distanced">
+        {products.map(({product, amount}, index) => (
```

Comment:
```suggestion
        {products.map(({productId, amount}, index) => (
```
Maybe clearer?

Reply:
the data structure is `products: [{product: 'some-id', amount: 42}]`, so deconstructing this would require `({product: productId, amount}, index)` in that case and I didn't find that particularly more readable, but either is fine.

I probably could also change the field name to be `productId` rather than `product` to clear things up?

Reply:
`product` is indeed the ID - note that the arrow function destructures the data already:

```js
(entry, index) => {...}
```
with `entry` an item of the `products` array.

so this destructures into:

```js
({product, amount}, index) => {...}
```